docs/adrs/v3-setup-dotnet.md

```diff
@@ -0,0 +1,48 @@
+# V3 setup-dotnet
+
+Date: 2021-06-30  
+Status: Proposed
+
+# Context
+- GitHub-hosted Ubuntu and Windows runners have .NET versions pre-installed in system directories
+  - Ubuntu:`/usr/share/dotnet`
+  - Windows:`C:\Program Files\dotnet`
+- V1 version of Action installs .NET to user's directory
+  - Ubuntu:`/home/runner/.dotnet`
+  - Windows: `C:\Users\runneradmin\AppData\Local\Microsoft\dotnet`
+- It means that action always download and install .NET version even if it is pre-installed. Also after using the action all pre-installed .NET versions are unavailable because `DOTNET_ROOT` is overriden to user directory.
+The behavior is different for macOS runners because pre-installation directory matches the one that is used by action. It means action can use pre-installed versions if it exists, which speeds up customer's workflow.
```

Comment:
The behavior is different for macOS runners because the pre-installation directory matches the one that is used by action. It means action can use pre-installed versions if they exist, which speeds up customer's workflow.